exercises/t_jubb/ex01_basics/src/find_words.cpp

```diff
@@ -0,0 +1,189 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include <fstream>
+#include <map>
+#include <set>
+#include <iomanip>
+#include <algorithm>
+#include <functional>
+
+using std::cout;
+using std::endl;
+
+
+std::vector<std::string> get_words(const std::string input_string, const int min_len) {
+  /*
+    Splits input_string up into its components words by finding a character, and then
+    iterating through characters until a non-alphabetic character is found.
+
+    @param input_string a string with spaces.
+    @param min_len, minimum lenth of word to store.
+    @return a vector of strings (the words).
+  */
+  std::vector<std::string> words;
+  
+  int index      = -1;
+  int word_start = 0;
+  int word_end   = 0;
+  int word_len   = 0;
+
+  for ( std::string::const_iterator i = input_string.begin(); i <= input_string.end(); i++ ) {
```

Comment:
The slightly more modern way to write this would be
```
for ( auto iter = std::cbegin(input_string); iter != std::cend(input_string); ++iter)
```
`auto` let's the compiler deduce the type rather than you having to spell it out.